WordPress/gutenberg
# Reusable blocks: Rename to 'Patterns' and add option to also add a non-synced Pattern

## What?

Renames the Reusable blocks to Patterns. Also adds the option to convert a block or collection of blocks to a non-synced Pattern. This then behaves as imported Patterns do in that once inserted they can be edited completely independently of any other instances of the Pattern. The synced version of the Pattern works as the existing Reusable blocks do in that editing one instance updates all instances.

## Why?
#50028 requires the ability of users to add new patterns and edit/duplicate existing patterns from the UI as part of the new Pattern Library functionality in the site editor. It made sense to reuse the `wp_block` CPT to implement this and to merge with Reusable blocks.

## How?
Adds a `wp_block.sync_status` post meta property to the `wp_block` CPT.

## N.B. - This PR only renames Reusable block in the UI, and the minimal critical places in code, in order to keep the PR as small and reviewable as possible. Follow-up PRs will rename all references in the code to reusable block to avoid future confusion. 

## Testing Instructions

- Before switching to PR branch add some reusable blocks
- Switch to branch and make sure the existing reusable blocks still work as expected
- Go to `/wp-admin/edit.php?post_type=wp_block` and check that the existing blocks have `Sync Status` set to `Synced` in the right post properties panel
- Add a new post and add a block or collection of blocks
- Select the block(s) and in the block toolbar overflow menu select the `Create a pattern` menu option
- The sync toggle will be off by default  - leave it that way and add the new pattern
- In the block inspector search for the new pattern name and insert
- Check that inserted blocks are independent/stand-alone blocks and not a synced reusable block
- Repeat the above but toggle sync on and make sure the saved block behaves the same as the existing reusable blocks
- Go to `/wp-admin/edit.php?post_type=wp_block` and make sure the synced and unsynced Patterns show. Edit Patterns from this page and check that toggling the sync status in right panel works as expected

## Screenshots or screencast 

https://github.com/WordPress/gutenberg/assets/3629020/86d4b5c0-abbc-4a4b-afc2-9f444efe74bf
